bin/hive_handling.dart

```diff
@@ -0,0 +1,196 @@
+          .add(HiveProduct(receiptName: name!, eanName: eanName!));
+      print('Product added!');
+      _countProducts(hiveProducts);
+    } else {
+      print('Product not added!');
+    }
+  }
+}
+
+/// Reads all products from the storage.
+void _readAllProducts(Box<HiveProduct> hiveProducts) {
+  _countProducts(hiveProducts);
+  print('All products:');
+  _printProducts(products: hiveProducts);
+}
+
+/// Prints Hive products.
+void _printProducts(
+    {required Box<HiveProduct> products, Iterable<dynamic>? filteredProducts}) {
+  if (filteredProducts == null) {
+    for (var product in products.keys) {
+      _printSingleProduct(products: products, product: product);
+    }
+  } else {
+    for (var product in filteredProducts) {
+      _printSingleProduct(products: products, product: product);
+    }
+  }
+}
+
+/// Prints single Hive product.
+void _printSingleProduct(
+    {required Box<HiveProduct> products, required dynamic product}) {
+  print('\t#$product: ${products.get(product)}');
+}
+
+/// Search by keyword in the storage.
+void _searchByKeyword(Box<HiveProduct> hiveProducts) {
+  print('Enter a keyword:');
+  var keyword = stdin.readLineSync();
+
+  if (keyword.isNotNullOrEmpty()) {
+    print('You entered: $keyword');
+
+    var filteredProducts = hiveProducts.keys.where((key) => hiveProducts
+        .get(key)!
+        .receiptName
+        .toLowerCase()
+        .contains(keyword!.toLowerCase()));
+
+    var amount = filteredProducts.length;
+
+    print('Found $amount ${amount == 1 ? 'product' : 'products'}:');
+
+    _printProducts(products: hiveProducts, filteredProducts: filteredProducts);
+  }
+}
+
+/// Updates a product in the storage.
+Future<void> _updateProduct(Box<HiveProduct> hiveProducts) async {
+  print('Enter the order number of the product:');
+
+  var orderNumber = stdin.readLineSync();
+
+  if (orderNumber.isNotNullOrEmpty()) {
+    var orderNumberAsInt = int.tryParse(orderNumber!);
+
+    var product = hiveProducts.get(orderNumberAsInt!);
+
+    if (product != null) {
+      print('Product: $product');
+
+      print('Enter the new receipt name of the product:');
+      var name = stdin.readLineSync();
+
+      print('Enter the new EAN name of the product:');
+      var eanName = stdin.readLineSync();
+
+      if (name.isNotNullOrEmpty() && eanName.isNotNullOrEmpty()) {
+        print('You entered: $name, $eanName');
+        print('Do you want to update this product? (y/n)');
+        var input = stdin.readLineSync();
+
+        if (input == 'y') {
+          await hiveProducts.put(orderNumberAsInt,
+              HiveProduct(receiptName: name!, eanName: eanName!));
+          print('Product updated!');
+          _countProducts(hiveProducts);
+        } else {
+          print('Product not updated!');
+        }
+      }
+    } else {
+      print('Product not found!');
+    }
+  }
+}
+
+/// Delete a product from the storage.
+Future<void> _deleteProduct(Box<HiveProduct> hiveProducts) async {
```

Comment:
The delete & update methods could be refactored somehow, they use the same order number logic in the beginning of the methods.

Reply:
I made some refactoring but I think there's still a bit more to do.

Reply:
I just made some more refactoring. I think it's ready now.